### PR TITLE
MF-309 Add support for a temporary config in localStorage

### PR DESCRIPTION
**Pull request by brandones**
Part of https://issues.openmrs.org/browse/MF-309

### CODE REVIEW

File: src/module-config/module-config.test.ts [hunk 1]
```diff
@@ -61,6 +61,29 @@ describe("defineConfigSchema", () => {
...
+
+  it("logs an error if any key does not include a default", () => {
+    const schema = {
+      foo: { bar: { description: "lol idk" } }
+    };
+    Config.defineConfigSchema("mod-mod", schema);
+    expect(console.error).toHaveBeenCalledWith(
+      expect.stringMatching(/mod-mod.*foo\.bar[\s\S]*default/)
+    );
+  });
```
Comment (brandones) [inline src/module-config/module-config.test.ts:86]: Unrelated to the temporary config thing.

File: src/module-config/module-config.ts [hunk 2]
```diff
@@ -99,6 +153,32 @@ function validateConfigSchema(
...
+          ].includes(k)
+        )
+      ) {
+        console.error(
+          `${moduleName} has bad config schema definition for key '${thisKeyPath}'. ${updateMessage}.` +
+            `\n\nIf you're the maintainer: all config elements must have a default. Received ${JSON.stringify(
+              schema[key]
+            )}`
+        );
+      }
```
Comment (brandones) [inline src/module-config/module-config.ts:181]: Unrelated to the temporary config thing.

**Review by sachin-0102 — APPROVED** [review]

**Comment by brandones** [comment]
Hi @sachin-0102 ! Thanks for reviewing—who are you?

**Comment by sachin-0102** [comment]
Hi @brandones. I am presently doing my B.tech. I wanted to contribute to OpenMRS and this was the first time I came accross this kind of issue, so I studied about it. and I also ran some tests on my computer and I thought what you did was correct so I reviewed it.

**Comment by brandones** [comment]
@sachin-0102  Cool! Welcome to the community! Are you on the Slack? Talk?

**Comment by sachin-0102** [comment]
I'm on Slack @brandones and I look forward to contributing to OpenMRS.

**Pull request merged**